client-lib/src/main/java/com/ibm/ws/repository/connections/RepositoryConnectionList.java

```diff
@@ -273,10 +274,23 @@ public void setUserAgent(String userAgent) {
     }
 
     /**
+     * Returns all resources which match the supplied set of FilterPredicate objects.
+     */
+    public Collection<RepositoryResource> getMatchingResources(final FilterPredicate... predicate) throws RepositoryBackendException {
+        Collection<RepositoryResource> resources = cycleThroughRepositories(new RepositoryInvoker<RepositoryResource>() {
```

Comment:
Further down, you used `DuplicatePolicy.ALLOW_DUPLICATES`

`cycleThroughRepositories` will discard duplicates, use `cycleThroughRepositoriesWithDupes` if you want to keep the duplicates from different repositories. (You probably don't want to keep duplicates but your usage of `ALLOW_DUPLICATES` elsewhere makes the intention unclear)